[DOCUMENTATION][feature/otx] update explanation templates. Introduce explanation section for classification task. (#1467)
open-edge-platform/training_extensions: docs/source/explanation/Main_algorithms/multi_class_classification.rst

Multi-class Classification
=========

*********
Dataset format
*********

We support a commonly used format for classification tasks: `imagenet <https://www.image-net.org/>`_ class folder format.
This format has the following structure:

::

    data
    ├── train
        ├── class 0
            ├── 0.png
            ├── 1.png
            ...
            └── N.png
        ├── class 1
            ├── 0.png
            ├── 1.png
            ...
            └── N.png
        ...
        └── class N
            ├── 0.png
            ├── 1.png
            ...
            └── N.png
    └── val
        ...

*********
Backbones
*********

> We have the following supported ready-to-use templates:

Reviewer: ```suggestion
We support the following ready-to-use templates:
```

+---------------------------------------------------+------------------------+---------------------+-----------------+
| Custom_Image_Classification_EfficinetNet-B0       | EfficientNet-B0        | 0.81                | 4.09            |
+---------------------------------------------------+------------------------+---------------------+-----------------+
| Custom_Image_Classification_EfficientNet-V2-S     | EfficientNet-V2-S      | 5.76                | 20.23           |
+---------------------------------------------------+------------------------+---------------------+-----------------+

The templates can be found `here <https://github.com/openvinotoolkit/training_extensions/blob/0d98bcd21d5e441516b8ec06949bc84870102b3f/otx/algorithms/classification/configs>`_.

Besides this, we support public backbones from `torchvision <https://pytorch.org/vision/stable/index.html>`_, `pytorchcv <https://github.com/osmr/imgclsmob>`_ and `OpenVino Model Zoo <https://github.com/openvinotoolkit/open_model_zoo>`_.
Please, refer to `tutorial <N/A>`_ how to run public backbones.

To see which public backbones are available for the task, the following command can be executed:

>    $ otx find --backbone {torchvision, pytorchcv}

Reviewer: ```suggestion
.. codeblock::

   $ otx find --backbone {torchvision, pytorchcv}
 
```

*********
Supervised Incremental Learning
*********

> We solve the multi-class classification problem in a common fashion based on the feature extractor backbone and classifier head that predicts the probability of each label presented on the image.

Reviewer: ```suggestion
We solve the multi-class classification problem in a common fashion: based on the feature extractor backbone and classifier head that predicts the probability of each label presented on the image.
```

For supervised training we use the following algorithms components:

- ``Augmentations``: Besides basic augmentations like random flip and random rotate, we use `Augmix <https://arxiv.org/abs/1912.02781>`_. This advanced type of augmentation helps to significantly expand the training distribution.

- ``Optimizer``: `Sharpness Aware Minimization (SAM) <https://arxiv.org/abs/2209.06585>`_. Wrapper upon the SGD that helps to achieve better generalization minimizing simultaneously loss value and loss sharpness.

> - ``Learning rate schedule``: `Cosine Annealing <https://arxiv.org/abs/1608.03983v5>`_ . It is a common learning rate scheduler that tends to work well on average for this task on variety of different datasets.

Reviewer: ```suggestion
- ``Learning rate schedule``: `Cosine Annealing <https://arxiv.org/abs/1608.03983v5>`_. It is a common learning rate scheduler that tends to work well on average for this task on variety of different datasets.
```

> - ``Loss function``: We use `Influence-Balanced Loss <https://arxiv.org/abs/2110.02444>`_. This is a balancing training method that helps us to solve the imbalanced data problem.

Reviewer: ```suggestion
- ``Loss function``:  `Influence-Balanced Loss <https://arxiv.org/abs/2110.02444>`_. This is a balancing training method that helps us to solve the imbalanced data problem.
```

> Additionaly, we use `No Bias Decay (NBD) <https://arxiv.org/abs/1812.01187>`_ technique and **early stopping** to add adaptability to the training pipeline and prevent overfitting.

Reviewer: I'd propose to formulate this paragraph as a list continuation

> To further enhance the performance of the algorithm in case when we have a small number of data we use `Supervised Contrastive Learning <https://arxiv.org/abs/2004.11362>`_. More concretely, we train a model with two heads: classification head with Influence-Balanced Loss and SupCon head with `Barlow Twins loss <https://arxiv.org/abs/2103.03230>`_

Reviewer: ```suggestion
To further enhance the performance of the algorithm in case when we have a small number of data we use `Supervised Contrastive Learning <https://arxiv.org/abs/2004.11362>`_. More concretely, we train a model with two heads: classification head with Influence-Balanced Loss and SupCon head with `Barlow Twins loss <https://arxiv.org/abs/2103.03230>`_.
```

> In the table below the top-1 accuracy on some academic datasets is presented. The results were obtained on our templates without any changes. We use 240x240 image resolution, for other hyperparameters, please, refer to the related template. We train all models on 1 GPU Nvidia GeForce GTX3090.

Reviewer 2: Why the resolution is 240? May be 224?
